src/agent/coverage/src/region.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::ops::Range;
+
+use serde::{Deserialize, Serialize};
+
+/// A non-overlapping set of regions of program data.
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+#[serde(transparent)]
+pub struct RegionIndex<R> {
```

Comment:
The impl of this type deserves unit tests. I will add those when I update it to support overlapping and duplicate regions.